packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+
+      const codesRow = document.createElement("td");
+      codesRow.setAttribute("ID", `problemCodes_${entryIndex}`);
+
+      const problemName = problemNameCell.childNodes[0].nodeValue;
+
+      const entries = Array.from(problemListComponent.getElementsByTagName("entry"));
+
+      const entry = entries.find(entry => {
+        const originalText = entry.getElementsByTagName("originalText")[0];
+        return originalText.childNodes[0].nodeValue === problemName;
+      });
+
+      if (entry) {
+        const value = entry.getElementsByTagName("value")[0];
+
+        const code = value.getAttribute("code");
+        const codeSystemName = value.getAttribute("codeSystemName");
+        const displayName = value.getAttribute("displayName");
+
+        const newCode = createCodeElement(code, codeSystemName, displayName, document);
+        codesRow.appendChild(newCode);
+
+        const translations = Array.from(value.getElementsByTagName("translation"));
+
+        translations.forEach(translation => {
+          const code = translation.getAttribute("code");
+          const codeSystemName = translation.getAttribute("codeSystemName");
+          const displayName = translation.getAttribute("displayName");
+
+          const newCode = createCodeElement(code, codeSystemName, displayName, document);
+          codesRow.appendChild(newCode);
+        });
+
+        tableCells.forEach(cell => {
+          if (cell.getAttribute("ID")?.includes("problemName")) {
+            const parentNode = cell.parentNode;
+            parentNode?.insertBefore(codesRow, cell.nextSibling);
+          }
+        });
+      }
+    }
+  }
+};
+
+const createCodeElement = (
+  code: string | null,
+  codeSystemName: string | null,
+  displayName: string | null,
+  document: Document
+) => {
+  const newCode = document.createElement("div");
+
+  const label = document.createElement("span");
+  label.setAttribute("class", "span_label");
+  label.appendChild(document.createTextNode(`${codeSystemName}: ${code}`));
+  newCode.appendChild(label);
+
+  newCode.appendChild(document.createElement("br"));
+
+  const display = document.createElement("span");
+  display.appendChild(document.createTextNode(`${displayName}`));
+  newCode.appendChild(display);
+
+  newCode.appendChild(document.createElement("br"));
+  newCode.appendChild(document.createElement("br"));
+
+  return newCode;
+};
+
+const createFileName = ({
+  cxId,
+  patientId,
+  resources,
+  dateFrom,
+  dateTo,
+}: {
+  cxId: string;
+  patientId: string;
+  resources?: ResourceTypeForConsolidation[];
+  dateFrom?: string;
+  dateTo?: string;
+}): string => {
+  const MEDICAL_RECORD_KEY = "MR";
+  let fileName = `${cxId}/${patientId}/${cxId}_${patientId}_${MEDICAL_RECORD_KEY}`;
+
+  if (resources) {
+    fileName = `${fileName}_${resources.toString()}`;
+  }
+
+  if (dateFrom) {
+    fileName = `${fileName}_${dateFrom}`;
+  }
+
+  if (dateTo) {
+    fileName = `${fileName}_${dateTo}`;
+  }
```

Comment:
what if they're both present? afaik we'll need to handle this case like `fileName = `${fileName}_${dateFrom}_to_${dateTo}`;`

Reply:
That's why I prefer to avoid variables at all cost and have single flow code. Like:

```ts
  const fileName = (suffix?: string) => {
    return `${cxId}/${patientId}/${cxId}_${patientId}_${MEDICAL_RECORD_KEY}${suffix ?? ""}.xml`;
  };
  if (resources) return fileName(resources.toString());
  if (dateFrom) return fileName(dateFrom);
  if (dateTo) return fileName(dateTo);
  return fileName();
```

But in this case, since I assume we want to have diff files for each request so we can keep them around and the cx still reference them, maybe we could create the file with a UUID at the end, and store the information about how it was generated on its S3 metadata?

Reply:
Yea if they are both true then the first one will be `${fileName}_${dateFrom}` and then the next one will append to 
`${fileName}_${dateFrom}_${dateTo}`

Reply:
How will we know which date it is if we find a file like this one? `<filename>_2023-10-01T02:00:00Z.xml`? Date from or date to?